src/api/DesktopAgent.ts

```diff
@@ -114,22 +114,46 @@ export interface DesktopAgent {
   broadcast(context: Context): void;
 
   /**
-   * Raises an intent to the desktop agent to resolve.
+   * Raises a specific intent for resolution against apps registered with the desktop agent.
+   *
+   * The desktop agent will resolve the correct app to target based on the provided intent name and context data. If multiple matching apps are found, the user may be presented with an app picker.
+   * Alternatively, the specific app to target can also be provided. A list of valid target applications can be retrieved via `findIntent`.
+   *
+   * Returns an `IntentResolution` object with details of the app that was selected to respond to the intent.
+   *
+   * If a target app for the intent cannot be found with the criteria provided, an `Error` with a string from the `ResolveError` enumeration is returned.
+   *
    * ```javascript
-   * //Find apps to resolve an intent to start a chat with a given contact
+   * // raise an intent for resolution by the desktop agent
+   * // a resolver UI will be displayed if more than one application can resolve the intent
+   * await fdc3.raiseIntent("StartChat", context);
+   * // or find apps to resolve an intent to start a chat with a given contact
    * const appIntent = await fdc3.findIntent("StartChat", context);
-   * //use the returned AppIntent object to target one of the returned chat apps with the context
+   * // use the returned AppIntent object to target one of the returned chat apps by name
    * await fdc3.raiseIntent("StartChat", context, appIntent.apps[0].name);
-   * //or use one of the AppMetadata objects returned in the AppIntent object's 'apps' array
-   * await fdc3.raiseIntent("StartChat", context, appMetadata);
+   * // or use one of the AppMetadata objects returned in the AppIntent object's 'apps' array
+   * await fdc3.raiseIntent("StartChat", context, appIntent.apps[0]);
    * ```
    */
   raiseIntent(intent: string, context: Context, app?: TargetApp): Promise<IntentResolution>;
 
   /**
-   * Raises a context to the desktop agent to resolve with one of the possible Intents for that context.
+   * Finds and raises an intent against apps registered with the desktop agent based purely on the type of the context data.
+   *
+   * The desktop agent will first resolve to a specific intent based on the provided context, displaying a resolver UI if more than one intent is available for specified context. It will then resolve to a specific app to handle the selected intent and specified context.
```

Comment:
```suggestion
   * The desktop agent SHOULD first resolve to a specific intent based on the provided context if more than one intent is available for the specified context. This MAY be achieved by displaying a resolver UI. It SHOULD then resolve to a specific app to handle the selected intent and specified context. 
```